STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/az_ulib_dm_interface.c

```diff
@@ -0,0 +1,133 @@
+// See LICENSE file in the project root for full license information.
+
+/********************************************************************
+ * This code was auto-generated from dm v2 DL.
+ *
+ * Implement the code under the concrete functions.
+ *
+ ********************************************************************/
+
+#include "_az_ulib_dm_interface.h"
+#include "az_ulib_capability_api.h"
+#include "az_ulib_descriptor_api.h"
+#include "az_ulib_ipc_api.h"
+#include "az_ulib_result.h"
+#include "az_ulib_dm_api.h"
+#include "dm_1_model.h"
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+static az_result dm_1_install_concrete(az_ulib_model_in model_in, az_ulib_model_out model_out)
+{
+  (void)model_out;
+  const dm_1_install_model_in* const in = (const dm_1_install_model_in* const)model_in;
+  return az_ulib_dm_install(in->package_name);
+}
+
+static az_result dm_1_install_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON to model_in_span.
+    az_json_reader jr;
+    dm_1_install_model_in install_model_in = { 0 };
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_init(&jr, model_in_span, NULL));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+    {
+      if (az_json_token_is_text_equal(&jr.token, AZ_SPAN_FROM_STR(DM_1_INSTALL_PACKAGE_NAME_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+        install_model_in.package_name
+            = az_span_create(az_span_ptr(jr.token.slice), az_span_size(jr.token.slice));
+      }
+      AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    }
+    AZ_ULIB_THROW_IF_AZ_ERROR(AZ_ULIB_TRY_RESULT);
+
+    // Call.
+    AZ_ULIB_THROW_IF_AZ_ERROR(dm_1_install_concrete((az_ulib_model_in)&install_model_in, NULL));
+
+    // Marshalling model_out to JSON.
+    *model_out_span = az_span_create_from_str("{}");
+  }
+  AZ_ULIB_CATCH(...) {}
+
+  return AZ_ULIB_TRY_RESULT;
+}
+
+static az_result dm_1_uninstall_concrete(az_ulib_model_in model_in, az_ulib_model_out model_out)
+{
+  (void)model_out;
+  const dm_1_uninstall_model_in* const in = (const dm_1_uninstall_model_in* const)model_in;
+  return az_ulib_dm_uninstall(in->package_name);
+}
+
+static az_result dm_1_uninstall_span_wrapper(az_span model_in_span, az_span* model_out_span)
+{
+  AZ_ULIB_TRY
+  {
+    // Unmarshalling JSON to model_in.
+    az_json_reader jr;
+    dm_1_uninstall_model_in uninstall_model_in = { 0 };
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_init(&jr, model_in_span, NULL));
+    AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
+    while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+    {
+      if (az_json_token_is_text_equal(&jr.token, AZ_SPAN_FROM_STR(DM_1_UNINSTALL_PACKAGE_NAME_NAME)))
+      {
+        AZ_ULIB_THROW_IF_AZ_ERROR(az_json_reader_next_token(&jr));
```

Comment:
Will we be replacing all if((result = function(...))) statements with this macro?

Reply:
Using this macros is up to the developer. I don't mind if you use it or not. This TRY/CATCH is a structured alternative to returns in the meddle of the code that I would like to avoid.